pkg/readiness/objset.go

```diff
@@ -27,9 +27,39 @@ type objKey struct {
 	namespacedName types.NamespacedName
 }
 
-type objSet map[objKey]struct{}
-
 func (k objKey) String() string {
 	return fmt.Sprintf("%s [%s]", k.namespacedName.String(), k.gvk.String())
+}
+
+type objData struct {
+	retries int
+}
+
+type objDataMutator func(*objData) *objData
+
+func setRetriesFromFlag(o *objData) *objData {
+	o.retries = *readinessRetries
+	return o
+}
+
+func mutatedObjData(muts ...objDataMutator) objData {
+	out := objData{}
+	for _, m := range muts {
+		m(&out)
+	}
+	return out
+}
+
+type objSet map[objKey]objData
 
+func (o objSet) decrementOrDelete(key objKey) bool {
+	val := o[key]
+	if val.retries > 0 {
```

Comment:
We should set this so a `retries` value of `-1` always returns `false`

Reply:
done.  I'll add a test for this as well